packages/core/src/payment/strategies/stripe-upe/stripe-upe-payment-strategy.ts

```diff
@@ -356,35 +356,16 @@ export default class StripeUPEPaymentStrategy implements PaymentStrategy {
             stripeConnectedAccount,
         );
 
-        let appearance: StripeUPEAppearanceOptions | undefined;
-
-        if (style) {
-            const styles = style;
-
-            appearance = {
-                variables: {
-                    colorPrimary: styles.fieldInnerShadow,
-                    colorBackground: styles.fieldBackground,
-                    colorText: styles.labelText,
-                    colorDanger: styles.fieldErrorText,
-                    colorTextSecondary: styles.labelText,
-                    colorTextPlaceholder: styles.fieldPlaceholderText,
-                    colorIcon: styles.fieldPlaceholderText,
-                },
-                rules: {
-                    '.Input': {
-                        borderColor: styles.fieldBorder,
-                        color: styles.fieldText,
-                        boxShadow: styles.fieldInnerShadow,
-                    },
-                },
-            };
-        }
+        const {
+            config: { getStoreConfig },
+        } = this._store.getState();
+        const features = getStoreConfig()?.checkoutSettings.features;
+        const experiment = features && features['CHECKOUT-6879.enable_floating_labels'];
```

Comment:
We'll shortly alter the condition to read it from the checkout configuration.
Therefore, the best way of moving forward is to use this helper function in `checkout-js`.
https://github.com/bigcommerce/checkout-js/blob/master/packages/core/src/app/common/utility/isFloatingLabelEnabled.ts
I think you may pass all of these UI properties from `checkout-js` while keeping the logic here.

Reply:
Same as above, I see the experiment is used to get styles from `getStripeCustomStyles` function. We should have this function in checkout-js if needed.

Reply:
Hey @animesh1987 and @bc-peng, I made changes, and it's ready for another review ♻️